source/extensions/quic_listeners/quiche/envoy_quic_proof_source_base.cc

```diff
@@ -0,0 +1,75 @@
+#include "extensions/quic_listeners/quiche/envoy_quic_proof_source_base.h"
+
+#pragma GCC diagnostic push
+
+// QUICHE allows unused parameters.
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include "quiche/quic/core/quic_data_writer.h"
+
+#pragma GCC diagnostic pop
+
+#include "extensions/quic_listeners/quiche/envoy_quic_utils.h"
+
+namespace Envoy {
+namespace Quic {
+
+void EnvoyQuicProofSourceBase::GetProof(const quic::QuicSocketAddress& server_address,
+                                        const quic::QuicSocketAddress& client_address,
+                                        const std::string& hostname,
+                                        const std::string& server_config,
+                                        quic::QuicTransportVersion /*transport_version*/,
+                                        quiche::QuicheStringPiece chlo_hash,
+                                        std::unique_ptr<quic::ProofSource::Callback> callback) {
+  quic::QuicReferenceCountedPointer<quic::ProofSource::Chain> chain =
+      GetCertChain(server_address, client_address, hostname);
+
+  size_t payload_size = sizeof(quic::kProofSignatureLabel) + sizeof(uint32_t) + chlo_hash.size() +
+                        server_config.size();
+  auto payload = std::make_unique<char[]>(payload_size);
+  quic::QuicDataWriter payload_writer(payload_size, payload.get(),
+                                      quiche::Endianness::HOST_BYTE_ORDER);
```

Comment:
Quiche depends on boringssl which only supports little-endian architectures. Since ARM is little endian, this will do the same thing on ARM as on x86. (The QuicDataWriter uses QuicheEndianImpl::HostToNet32 under the hood.)

Reply:
Expanding on my previous message: This only matters for Google QUIC Crypto, which has assumptions of little endian baked in throughout it. Given that this code already only runs on little endian architectures and Google QUIC Crypto is destined to die soon, I don't see any reason to worry about architecture endianness.

Reply:
Huh, I always though that ARM was big-endian, but a quick look at wikipedia says it is bi-endian, but usually run as little-endian. TIL.